processing/src/main/java/io/druid/query/lookup/LookupSnapshotTaker.java

```diff
@@ -0,0 +1,94 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.lookup;
+
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.google.common.io.Files;
+import com.metamx.common.ISE;
+import com.metamx.common.StringUtils;
+import com.metamx.common.logger.Logger;
+import io.druid.guice.annotations.Json;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+
+public class LookupSnapshotTaker
+{
+  private static final Logger LOGGER = new Logger(LookupSnapshotTaker.class);
+
+  private final ObjectMapper objectMapper;
+  private final File persistDirectory;
+  private final File persistFile;
+
+
+  public LookupSnapshotTaker(
+      final @Json ObjectMapper jsonMapper,
+      String persistDirectory
+  )
+  {
+    this.objectMapper = jsonMapper;
+    this.persistDirectory = persistDirectory == null
+                            ? Files.createTempDir()
+                            : new File(persistDirectory);
+    if (!this.persistDirectory.exists()) {
+      this.persistDirectory.mkdirs();
+    }
+    if (!this.persistDirectory.isDirectory()) {
+      throw new ISE("Can only persist to directories, [%s] wasn't a directory", persistDirectory);
+    }
+    this.persistFile = new File(persistDirectory, "lookupSnapshot.data");
+  }
+
+  public synchronized List<LookupBean> pullExistingSnapshot()
+  {
+    List<LookupBean> lookupBeanList;
+    try {
+      if (!persistFile.isFile()) {
+        LOGGER.warn("could not find any snapshot file under working directory [%s]", persistDirectory);
+        return Collections.emptyList();
+      }
+      lookupBeanList = objectMapper.reader(new TypeReference<List<LookupBean>>()
+      {
+      }).readValue(Files.toString(persistFile, StringUtils.UTF8_CHARSET));
+      return lookupBeanList;
+    }
+    catch (IOException e) {
+      throw new ISE(e, "Exception during reading lookups from [%s]", persistFile.getAbsolutePath());
+    }
+  }
+
+  public synchronized void takeSnapshot(List<LookupBean> lookups)
+  {
+    try {
+      String lookupsSnapshot = objectMapper.writeValueAsString(lookups);
+      Files.write((StringUtils.toUtf8(lookupsSnapshot)), persistFile);
```

Comment:
An `ObjectMapper` can actually both read and write directly from a File and it does this in a streaming manner, so I would recommend doing that instead of materializing into a String.  It's not a big deal, though.